tools/model_tools/src/openvino/model_zoo/download_engine/cache.py

```diff
@@ -91,6 +93,37 @@ def put(self, hash, path):
                 staging_path.unlink()
 
 
+class Checksum(base.TaggedBase):
+    types = {}
+
+    @classmethod
+    def deserialize(cls, checksum):
+        if isinstance(checksum, str):
+            checksum = {'$type': 'sha384', 'value': checksum}
+        return super().deserialize(checksum)
+
+
+class ChecksumSHA384(Checksum):
+    RE_SHA384SUM = re.compile(r'[0-9a-fA-F]{96}')
+
+    def __init__(self, value):
+        self.algo = hashlib.sha384()
+        self.value = value
+
+    @classmethod
+    def deserialize(cls, checksum):
+        sha384_str = validation.validate_string('"sha384"', checksum['value'])
+        if not cls.RE_SHA384SUM.fullmatch(sha384_str):
+            raise validation.DeserializationError(
+                '"sha384": got invalid hash {!r}'.format(sha384_str))
+
+        sha384 = bytes.fromhex(sha384_str)
+        return cls(sha384)
+
+
+Checksum.types['sha384'] = ChecksumSHA384
```

Comment:
Sometime you call hashing algorithms `algo` and sometimes `type`. I suggest being consistent :)

Reply:
Done